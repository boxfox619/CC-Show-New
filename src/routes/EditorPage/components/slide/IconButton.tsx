import * as React from 'react';
import styled from 'styled-components';

const IconButtonContainer = styled.img`
    font-size: 1.5rem;
    width: 0.6em;
    color: white;
    position: relative;
    margin: 3px 5px 0 5px;
    transition: transform 0.3s ease;
    &:hover {
        background-color: transparent;
        transform: rotate(90deg);
        cursor: pointer;
    }
`

interface OwnProps {
    icon: string
}

type Props = OwnProps & React.HTMLAttributes<HTMLDivElement>;

const IconButton: React.FC<Props> = (props: Props) => {
    const divProps = props as React.HTMLAttributes<HTMLDivElement>;
    return (
        <IconButtonContainer {...divProps} src={props.icon}/>
    )
}

export default IconButton;