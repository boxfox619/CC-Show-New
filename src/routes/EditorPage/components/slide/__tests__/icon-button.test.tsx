import * as React from 'react';
import * as Enzyme from 'enzyme';
import IconButton from '../IconButton';

describe('<IconButton />', () => {
  const mockFn = jest.fn();
  it('should match snapshot', () => {
    const wrapper = Enzyme.shallow(<IconButton icon="https://avatars0.githubusercontent.com/u/14067209?s=88&v=4" onClick={mockFn}/>);
    expect(wrapper).toMatchSnapshot();
  });
});