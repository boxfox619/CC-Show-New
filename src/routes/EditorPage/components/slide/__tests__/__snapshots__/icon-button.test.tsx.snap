// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconButton
    icon="https://avatars0.githubusercontent.com/u/14067209?s=88&v=4"
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "icon": "https://avatars0.githubusercontent.com/u/14067209?s=88&v=4",
      "onClick": [MockFunction],
      "src": "https://avatars0.githubusercontent.com/u/14067209?s=88&v=4",
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    font-size: 1.5rem;
    width: 0.6em;
    color: white;
    position: relative;
    margin: 3px 5px 0 5px;
    transition: transform 0.3s ease;
    &:hover {
        background-color: transparent;
        transform: rotate(90deg);
        cursor: pointer;
    }
",
        ],
      },
      "displayName": "styled.img",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "img",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "icon": "https://avatars0.githubusercontent.com/u/14067209?s=88&v=4",
        "onClick": [MockFunction],
        "src": "https://avatars0.githubusercontent.com/u/14067209?s=88&v=4",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    font-size: 1.5rem;
    width: 0.6em;
    color: white;
    position: relative;
    margin: 3px 5px 0 5px;
    transition: transform 0.3s ease;
    &:hover {
        background-color: transparent;
        transform: rotate(90deg);
        cursor: pointer;
    }
",
          ],
        },
        "displayName": "styled.img",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "img",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
